src/chat.cpp

```diff
@@ -313,22 +340,115 @@ u32 ChatBuffer::formatChatLine(const ChatLine& line, u32 cols,
 			// remaining_in_{in,out}put. Try to end the fragment
 			// on a word boundary.
 			u32 frag_length = 1, space_pos = 0;
-			while (frag_length < remaining_in_input &&
-					frag_length < remaining_in_output)
+
+			if(!m_cache_clickable_chat_weblinks)
+			{
+				while (frag_length < remaining_in_input &&
+						frag_length < remaining_in_output)
+				{
+					if (iswspace(line.text.getString()[in_pos + frag_length]))
+						space_pos = frag_length;
+					++frag_length;
+				}
+				if (space_pos != 0 && frag_length < remaining_in_input)
+					frag_length = space_pos + 1;
+
+				temp_frag.text = line.text.substr(in_pos, frag_length);
+				temp_frag.column = 0;
+				//temp_frag.bold = 0;
+				next_frags.push_back(temp_frag);
+				in_pos += frag_length;
+				text_processing = true;
+			}
+			// Unless weblinks are enabled. Then it's slightly more complex
+			else
 			{
-				if (iswspace(line.text.getString()[in_pos + frag_length]))
-					space_pos = frag_length;
-				++frag_length;
+				u32 http_pos = u32(std::wstring::npos);
+				text_processing = false;  // set FALSE at end of this block
+				bool halt_fragloop = false;
+				while(!halt_fragloop)
+				{
+					remaining_in_input = line.text.size() - in_pos;
+					frag_length = 0;
+					space_pos = 0;
+
+					// Note: unsigned(-1) on fail
+					http_pos = u32(line.text.getString().find(L"https://", std::size_t(in_pos)));
+					if(http_pos == u32(std::wstring::npos))
+						http_pos = u32(line.text.getString().find(L"http://", std::size_t(in_pos)));
+					if(http_pos != u32(std::wstring::npos))
+						http_pos -= in_pos;
+
+					while (frag_length < remaining_in_input &&
+							frag_length < remaining_in_output)
+					{
+						if (iswspace(line.text.getString()[in_pos + frag_length]))
+							space_pos = frag_length;
+						++frag_length;
+					}
+
+					// Http not in range, grab until space or EOL, halt as normal.
+					if(http_pos > remaining_in_output)  // sufficient unless screen is infinitely wide
+					{
+						halt_fragloop = true;
+						// force text processing on THIS frag
+						text_processing = true;
+					}
+					// At http, grab ALL until FIRST whitespace or end marker. loop.
+					// If at end of string, next loop will be empty string to mark end of weblink. This is intentional.
+					else if(http_pos == 0)
+					{
+						frag_length = 6;
+						// TODO? replace this with a safer (slower) regex whitelist?
+						wchar_t tempchar = line.text.getString()[in_pos+frag_length];
+						while (frag_length < remaining_in_input &&
+								!iswspace(tempchar) &&
+								tempchar != L'\'' &&
+								tempchar != L'\"' &&
+								tempchar != L')' &&
+								tempchar != L';'
```

Comment:
Yeah this part started getting out of hand. It would be more "secure" anyway to use a regex whitelist rather than delimiter, but I know that would use a lot more cpu.  I'd appreciate insight/opinions from devs about that option.